.envrc

```diff
@@ -0,0 +1 @@
+use nix
```

Comment:
The main problem with this seems to be that it doesn't require the user to confirm it if the `.envrc.private` file changed, see https://github.com/direnv/direnv/issues/348. Especially in Nixpkgs this considerable increases the attack surface, since we run so many update scripts, which could populate `.envrc.private` file.

Reply:
How is this different from changing shell.nix to add a shellHook or changing any code in nixpkgs?

Reply:
I guess at least the `shell.nix` is being code owned by the security team, so if somebody does something sketchy they'd be notified. Can't do the same here. Ping @NixOS/security for input

Reply:
Why not add .envrc.private than as well to the code owner list?

Reply:
Anyway. Not the hill, I want to die on.

Reply:
> Why not add .envrc.private than as well to the code owner list?

Unless I'm missing something, code owners only work when you commit the file to GitHub, which defeats the point of it :P

Anyways, I don't want to say No to this, just that I'm personally not sure about the security impact of it. So I'd say let's go with a more minimal solution without `.private` for now, which we can then still change in a backwards-compatible way later if necessary.